OpenRA.Game/Traits/Target.cs

```diff
@@ -204,6 +204,15 @@ public bool IsInRange(WPos origin, WDist range)
 			return Positions.Any(t => (t - origin).HorizontalLengthSquared <= range.LengthSquared);
 		}
 
+		public bool CenterIsInRange(WPos origin, WDist range)
```

Comment:
I'm not sure adding this here is the best idea, considering that checking the range at all is a bodge that will be removed when we add proper docking logic. Can we inline the check directly in `Resupply` instead?